include/swift/SIL/SILOpenedArchetypesTracker.h

```diff
@@ -44,12 +36,6 @@ class SILOpenedArchetypesTracker : public DeleteNotificationHandler {
   SILOpenedArchetypesTracker(SILOpenedArchetypesTracker &Tracker)
       : SILOpenedArchetypesTracker(Tracker.F, Tracker) {}
```

Comment:
Take a look at the push, it incurs new/delete operations but It compilers, fixes the warning and tests successfully